index.js

```diff
@@ -1,38 +1,255 @@
       cell.textContent = `${x},${y}`;  // saves the coordinates as a string value 'x,y'
-      cell.value = 0;//state of the cell
+      cell.value = player.gameBoard[x][y];//state of the cell
 
       //this function adds the click event to each cell
-      cell.addEventListener( 'click', (e) => {
-          let cell = e.target; // get the element clicked
-          console.log( cell.textContent) //display the coordinates in the console
-          cell.style.visibility = 'hidden';// this  means that the contents of the element will be invisible, but the element stays in its original position and size / try it clicking on any of the black cells (in your browser) and see whats happens
-          //cell.style.background ="purple"; //with this propertie you can change the background color of the clicked cell. try comment the line bellow and uncomment this line. Do not forget to save this file and refresh the borwser to see the changes
-      });
+      cell.addEventListener('click', (e) => {
+        if(activeBoard.contains(cell)) {
+       // change players and boards
+        [currentPlayer, opponent] = [opponent, currentPlayer];
+        [activeBoard, inactiveBoard] = [inactiveBoard, activeBoard];
+
+          turnPlayer.textContent = currentPlayer.name;
+          // set the element clicked
+          let cell = e.target; 
+
+          if (cell.value === 1) {
+            //condition for hit
+            alert("Hit!");
+            cell.style.background = "red";
+            cell.textContent = "ship!";
+            //opponent.shipCount--;
+            currentPlayer.shipCount--;
+            if (playerOne === opponent) {
+              //update of shipcount label content after hit
+              lives2.innerHTML -= 1;
+            } else {
+              lives1.innerHTML -= 1;
+            }
+          } else if (cell.value === 0) {
+            //condition for the failure
+            cell.style.background = "purple";
+            //cell.style.visibility = "hidden";
+            cell.textContent = "no ship!";
+            alert("Miss!!");
 
+            // confirm the continue of game    
+            continueGame = confirm("Would you like to continue?");
+            if (!continueGame) {
+            alert (`Stop Game by ${opponent.name}`);
+            }
+          }
+          if (currentPlayer.shipCount === 0) {
+            message = opponent.name;
+            alert(`Congratulations ${message}!! you win`);
+            console.log(`Congratulations ${message}!! you win`);
+          }
+        }
+        console.log(currentPlayer, opponent);
+      });
       li.appendChild(cell); //adding each cell into the row number x
     }
-
-     board_Player1.appendChild(li); //adding each row into the board
+  //adding each row into the board
+    if (player === playerOne) {
+      board_player1.appendChild(li);
+    } else if (player === playerTwo) {
+      board_player2.appendChild(li);
+    }
+  }
 }
```

Comment:
This function looks pretty long. It would be better splitting this function into several smaller functions.

Reply:
Please give me an example, which part of the code I can split into the functions?